tests/lib/rules/max-len.js

```diff
@@ -650,6 +734,204 @@ ruleTester.run("max-len", rule, {
                     column: 1
                 }
             ]
+        },
+
+        // https://github.com/eslint/eslint/issues/12213
+        {
+            code: "var jsx = (<>\n" +
+                  "  { /* this line has 38 characters */}\n" +
+                  "</>)",
+            options: [15, { comments: 37 }],
+            parserOptions: { ecmaFeatures: { jsx: true } },
+            errors: [
+                {
+                    messageId: "maxComment",
+                    data: { lineLength: 38, maxCommentLength: 37 },
+                    type: "Program",
+                    line: 2,
+                    column: 1
+                }
+            ]
+        },
+        {
+            code: "var jsx = (<>\n" +
+                  "\t\t{ /* this line has 40 characters */}\n" +
+                  "</>)",
+            options: [15, 4, { comments: 40 }],
+            parserOptions: { ecmaFeatures: { jsx: true } },
+            errors: [
+                {
+                    messageId: "maxComment",
+                    data: { lineLength: 44, maxCommentLength: 40 },
+                    type: "Program",
+                    line: 2,
+                    column: 1
+                }
+            ]
+        },
+        {
+            code: "var jsx = (<>\n" +
+                  "{ 38/* this line has 38 characters */}\n" +
+                  "</>)",
+            options: [15, { comments: 37 }],
```

Comment:
This could be `comments: 38` to show that line fails because it isn't a comment?

Also, it feels like some tests with an expression (`a && b`) + comment inside `{}` might be missing `ignoreTrailingComments: true` or `ignoreComments: true` to show that this doesn't count as a comment?

And I think (could be wrong) that there is no test where there is only a comment inside `{}`, but `{` and `}` are not on the same line so it doesn't count as a comment because of that.

Reply:
Sure!

Reply:
@mdjermanovic 
Humm.. what do you think about this case? A multi-line comment in JSXEmptyExpression should be treated as a code?
```jsx
var jsx = (<foo>
     {/* multiline comment
         without code */ }
</foo>);
```
so if then, Should I treat the case below as an invalid?
```
 {
     code: "var jsx = (<>\n" +
                  "  {/*  multi-line          comment\n" +
                  "                   but it is a code ? */ }\n"
                  "</>)",
    options: [20, { ignoreComments: true }],
    parserOptions: { ecmaFeatures: { jsx: true } }
 }
```

Reply:
@mdjermanovic 
> I think the decision to treat the whole { /* */ } as a comment only if everything incl. braces is on the same line is a good decision, since it's a minimal change in behavior.

Humm, I'm confusing 🤔
Maybe treating the whole {/* ... *} as a comment in this PR would be better? ( and change tag `fix` to `update`)

Reply:
Sorry for the confusion! Just wanted to say that I agree that only those `{/* */}` where everything is on the same line should be treated as comments, as you already implemented with the `parent.loc.start.line === parent.loc.end.line` condition :+1: